crates/types/src/traits/block_contents.rs

```diff
@@ -83,13 +89,16 @@ pub trait BlockPayload:
 /// # Panics
 /// If the VID computation fails.
 #[must_use]
-pub fn vid_commitment(encoded_transactions: &Vec<u8>) -> <VidScheme as VidSchemeTrait>::Commit {
+pub fn vid_commitment(
+    encoded_transactions: &Vec<u8>,
+    num_storage_nodes: usize,
```

Comment:
In the future I expect we'll need more VID params beyond `num_storage_nodes`. There's a good chance this arg will be replaced by a full on `&VidScheme` or whatever.

Reply:
Yeah, for now I guess we could create a static VidScheme on startup.

In the future we need to create a new VidScheme whenever we change the number of storage nodes. I guess right now that's in the Transactions task? That feels less than ideal to me, but I can't think of anything better.

Reply:
whatever we do in the future, I'm 99% certain we will *not* be constructing a `VidScheme` inside the `vid_commitment` function; these VID params will be needed elsewhere, too.

Reply:
In order for us to be able to pass the VidScheme around between nodes, it needs to implement `Serialize` and `Deserialize`. I think we need this to do things like include it in the `DAProposal`, wherein replicas need to vote on the commitment.

Can we (I) just derive it for `Advz` and make a PR? Or is it more complex than that? Or maybe we can offload this to later and just use the `num_committee_nodes` for now (or create a static one on startup).

Does this make sense? WDYT?